libhoney/transmission.py

```diff
@@ -92,23 +97,29 @@ def _send(self, ev):
             "body": resp.text,
             "error": "",
         }
-        try:
-            if self.block_on_response:
-                self.responses.put(response)
-            else:
+        if self.block_on_response:
+            self.responses.put(response)
+        else:
+            try:
                 self.responses.put_nowait(response)
-        except queue.Full:
-            pass
+            except queue.Full:
+                pass
 
     def close(self):
         '''call close to send all in-flight requests and shut down the
            senders nicely'''
         for i in range(self.max_concurrent_batches):
-            self.pending.put(None)
+            try:
+                self.pending.put(None, true, 10)
+            except queue.Full:
+                pass
         for t in self.threads:
             t.join()
         # signal to the responses queue that nothing more is coming.
-        self.responses.put(None)
+        try:
+            self.responses.put(None, true, 10)
```

Comment:
what are the extra args here?  to keep us from hanging forever if the responses queue is full?